target: 'serverless' on local build

I just spend one hour investigating why next build does not work locally then I realized I have target set 'serverless' in next.config.js.

Maybe it should be ignored for the local build.

```
Error: Cannot find module '/Users/steida/dev/este/.next/server/static/jAvZuFRlVjcCJxcCa5aNH/pages/_document'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:581:15)
    at Function.Module._load (internal/modules/cjs/loader.js:507:25)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
```

Just to clarify: `next build` alone seems to be working just fine with `target: 'serverless'` in `next.config.js`. Only after running `next start` it logs `Error: Cannot find module '_document`' at the moment you try to access `localhost:3000` in the browser.

--- a/errors/next-start-serverless.md
+++ b/errors/next-start-serverless.md
@@ -1,0 +1,9 @@
+# Using `next start` with `target` not set to `server`
+
+#### Why This Error Occurred
+
+Next.js can only handle running a server when the `target` is set to `server` (this is the default value). A serverless build, for instance, has no handler for requests–this is usually implemented by a hosting provider.
+
+#### Possible Ways to Fix It
+
+Use a different handler than `next start` when testing a serverless **production** build, otherwise just use `next dev`.

--- a/packages/next-server/server/next-server.ts
+++ b/packages/next-server/server/next-server.ts
@@ -47,7 +47,9 @@ export default class Server {
 
     // Only serverRuntimeConfig needs the default
     // publicRuntimeConfig gets it's default in client/index.js
-    const {serverRuntimeConfig = {}, publicRuntimeConfig, assetPrefix, generateEtags} = this.nextConfig
+    const {serverRuntimeConfig = {}, publicRuntimeConfig, assetPrefix, generateEtags, target} = this.nextConfig
+
+    if (process.env.NODE_ENV === 'production' && target !== 'server') throw new Error('Cannot start server when target is not server. https://err.sh/zeit/next.js/next-start-serverless')
 
     this.buildId = this.readBuildId()
     this.renderOpts = {

